### ISSUE TITLE
[DX] backdrop_process_states only works on lowest level of field in node form

### ISSUE THREAD

**Issue by bradbulger** [backdrop/backdrop-issues#3489]
In a node form array, the actual field array of type "#text" is located at something like "field_name[und][0][#value]". All the documentation of backdrop_process_states() describe it as working on the top level - ie setting `$form['field_myfield]['#states']`. That's how it works in Drupal 7 & 8.

I'm finding, though, that I have to go down to the text field - set `$form['field_myfield']['und'][0]['value']['#states']`. Setting at the initial field level has no or unexpected behavior. For instance, if the fields are inside a vertical tab fieldset, the whole fieldset gets set to "display:none;".

I am wondering if this is expected behavior? If it is, the documentation could be clearer about that. If not, then I suppose it's a bug.

This is in the Seven theme, if that's relevant. Backdrop 1.11.3.

**Comment by bradbulger** [comment]
What's proving to be a simpler workaround is to set the language code array to a `#type` of "container" if it isn't set already, and apply `#states` at that level.

**Comment by oadaeh** [comment]
I can confirm this behavior.
However, for me I did not need `[0]['#value']`. It worked fine with only `$form['field_myfield']['und']['#states']`

**Comment by jenlampton** [comment]
This is weird, and I don't think intended. Adding milestone candidate label.

**Comment by stpaultim** [comment]
If I understand it correctly, this issue may be blocking the release of the Conditional Fields module.
https://github.com/backdrop-contrib/conditional_fields/issues/2

**Comment by laryn** [comment]
Does the PR on this issue help here, too?
https://github.com/backdrop/backdrop-issues/issues/5157

**Comment by bradbulger** [comment]
I'll try to test that.

**Comment by bradbulger** [comment]
OK, I was able to recreate the original problem, and then applied the PR from #5157 and that fixed it, in that it worked as documented. I was able to define `$form['field_mytextfield']['#states']` and make visibility contingent on another field in the form.

**Comment by laryn** [comment]
Great! Are you able to glance over the code on that PR and give it a thumbs up if it makes sense to you? Then I think that PR would be mergable (I've already done a WFM on that PR).

**Comment by bradbulger** [comment]
yeah it seems fine

**Comment by klonos** [comment]
Well, #5157 has been merged, so all we need to do here is to confirm that this problem has been fixed with the 1.20.3 bug-fix release that's about to be cut 😉 

@bradbulger please test as soon as you get a chance, and report back. Thanks.

**Comment by bradbulger** [comment]
Yep, I pulled 1.20.2 and my test case failed, then pulled 1.20.3 and it is all better.

**Comment by klonos** [comment]
Thanks for taking the time to test and report back @bradbulger 🙏🏼  ...lets call this fixed then.
